src/providers/FusionAuthProvider.tsx

```diff
@@ -35,154 +37,194 @@ export type RedirectSuccess = (state: string) => void;
             Cookies.set('lastState', stateParam);
-            const code = await generatePKCE();
-            Cookies.set('codeVerifier', code.code_verifier);
-            const queryParams = {
-                client_id: props.clientID,
-                scope: props.scope ?? DEFAULT_SCOPE,
-                response_type: 'code',
-                redirect_uri: props.redirectUri,
-                code_challenge: code.code_challenge,
-                code_challenge_method: 'S256',
-                state: stateParam,
-            };
-            const fullUrl = generateUrl(FunctionType.register, queryParams);
+            const fullUrl = generateServerUrl(
+                ServerFunctionType.register,
+                props.registerPath,
+                {
+                    client_id: props.clientID,
+                    redirect_uri: props.redirectUri,
+                    scope: props.scope ?? DEFAULT_SCOPE,
+                    state: stateParam,
+                },
+            );
             window.location.assign(fullUrl);
         },
-        [generateUrl, props.clientID, props.redirectUri, props.scope],
+        [
+            generateServerUrl,
+            props.clientID,
+            props.redirectUri,
+            props.registerPath,
+            props.scope,
+        ],
     );
 
     useEffect(() => {
         const userCookie = Cookies.get('user');
         if (userCookie) {
-            setUser(JSON.parse(userCookie));
+            try {
+                setUser(JSON.parse(userCookie));
+            } catch {
+                // ignore errors
+            }
         }
     }, [setUser]);
 
-    const refreshToken = useCallback(async () => {
-        await fetch(`${props.serverUrl}/jwt-refresh`, {
-            method: 'POST',
-            headers: {
-                'content-type': 'application/json',
-            },
-            credentials: 'include',
-        });
-    }, [props.serverUrl]);
+    const refreshToken = async () => {
+        const accessTokenExpires = Cookies.get('access_token_expires');
+        const timeWindow =
+            props.accessTokenExpireWindow ?? DEFAULT_ACCESS_TOKEN_EXPIRE_WINDOW;
+        if (
+            accessTokenExpires === undefined ||
+            Number(accessTokenExpires) < Date.now() + timeWindow
+        ) {
+            await fetch(
+                generateServerUrl(
+                    ServerFunctionType.tokenRefresh,
+                    props.tokenRefreshPath,
+                ),
+                {
+                    method: 'POST',
+                    headers: {
+                        // some servers expect content-type, even with no body
+                        'content-type': 'text/plain',
+                    },
+                    credentials: 'include',
+                },
+            );
+        }
+    };
 
+    // TODO - known issue of side effects in useLayoutEffect()
     useLayoutEffect(() => {
+        console.log('in useLayoutEffect');
```

Comment:
I think we should remove console.log statements. If we want to add some debug logging maybe we can enable that via configuration

Reply:
agreed.  removed for now.